plugins/c8y_remote_access_plugin/src/lib.rs

```diff
@@ -155,38 +154,19 @@ async fn spawn_child(command: String, config_dir: &Utf8Path) -> miette::Result<(
     }
 }
 
-#[derive(Copy, Clone, Eq, PartialEq, Debug)]
-enum WsProtocol {
-    Ws,
-    Wss,
-}
-
-impl Display for WsProtocol {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        match self {
-            Self::Ws => "ws".fmt(f),
-            Self::Wss => "wss".fmt(f),
-        }
-    }
-}
-
 async fn proxy(command: RemoteAccessConnect, config: TEdgeConfig) -> miette::Result<()> {
-    let host = &config.c8y.proxy.client.host;
-    let port = config.c8y.proxy.client.port;
-    let protocol = config
+    let host = config
         .c8y
-        .proxy
-        .cert_path
-        .or_none()
-        .map_or(WsProtocol::Ws, |_| WsProtocol::Wss);
-    let url = build_proxy_url(protocol, host, port, command.key())?;
-    let client_config = config
         .http
-        .client_tls_config()
-        .map_err(|e| miette!("{e}"))?;
+        .or_config_not_set()
+        .into_diagnostic()?
+        .to_string();
+    let url = build_proxy_url(host.as_str(), command.key())?;
+    let jwt = Jwt::retrieve(&config)
+        .await
+        .context("Failed when requesting JWT from Cumulocity")?;
 
-    let proxy =
-        WebsocketSocketProxy::connect(&url, command.target_address(), client_config).await?;
+    let proxy = WebsocketSocketProxy::connect(&url, command.target_address(), jwt).await?;
```

Comment:
@jarhodes314  Do we know which ca certificates are used here (e.g. system or the custom c8y.root_cert_path setting?). Or would this handling be covered by your PR, https://github.com/thin-edge/thin-edge.io/pull/2953

Reply:
Good point. It'll almost certainly not be using `root_cert_path`, but I'll deal with it in my PR.